### PR TITLE
Added advanced configuration to the Ocean framework page

### PR DESCRIPTION
**Pull request by MatanHeledPort**
# Description

What - Added advanced configuration section to the Ocean framework docs. This includes proxy configuration and SSL configuration.
Why - To allow setting up ocean in environments with special requirements.
How -

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

### CODE REVIEW

File: docs/framework-guides/docs/deployment/helm.md [hunk 1]
```diff
@@ -62,3 +62,14 @@ the [integration configuration](../develop-an-integration/integration-configurat
 More information about the available event listener types and optional configurations can be found in
 the [event listeners](../framework/features/event-listener.md) guide.
 :::
+
+## Advanced configruation
+The Ocean framework supports [advanced configuration](../framework/advances-configuration.md) using environment variables. The Ocean Helm chart allows setting these variables using Helm parameters. This can be done either by:
```
Comment (MPTG94) [inline docs/framework-guides/docs/deployment/helm.md:67]: Either by OR???

File: docs/framework-guides/docs/framework/advances-configuration.md
Comment (MPTG94) [inline docs/framework-guides/docs/framework/advances-configuration.md:1]: The file should be called advnaced configuration. Not advances

File: docs/framework-guides/docs/framework/advances-configuration.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+---
+title: Advanced Configuration
+sidebar_label: ⚙️ Advanced Configuration
+sidebar_position: 3
+---
+
+# ⚙️ Advanced configuration
+The Ocean framework is based on a Python HTTP client called [HTTPX](https://www.python-httpx.org/). The HTTPX client provides out-of-the-box configuration capabilites, which allows configuring the client to work with:
+- Self-signed certificates
```
Comment (MPTG94) [inline docs/framework-guides/docs/framework/advances-configuration.md:9]:
```suggestion
- Self-signed certificates.
```

File: docs/framework-guides/docs/framework/advances-configuration.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+- Proxies
```
Comment (MPTG94) [inline docs/framework-guides/docs/framework/advances-configuration.md:10]:
```suggestion
- Proxies.
```

File: docs/framework-guides/docs/framework/advances-configuration.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+---
+title: Advanced Configuration
+sidebar_label: ⚙️ Advanced Configuration
+sidebar_position: 3
+---
+
+# ⚙️ Advanced configuration
+The Ocean framework is based on a Python HTTP client called [HTTPX](https://www.python-httpx.org/). The HTTPX client provides out-of-the-box configuration capabilites, which allows configuring the client to work with:
```
Comment (MPTG94) [inline docs/framework-guides/docs/framework/advances-configuration.md:8]: Add a line break between this paragraph and the list underneath it, just for markdown readability

File: docs/framework-guides/docs/framework/advances-configuration.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+ALL_PROXY=http://my-proxy.com:3333
+```
+
+### `NO_PROXY`
+
+`NO_PROXY` allows blacklisting certain addresses from being handled through a proxy. This vairable accepts a comma-seperated list of hostnames or urls.
+
+For example:
+```sh showLineNumbers
+NO_PROXY=http://127.0.0.1,google.com
```
Comment (MPTG94) [inline docs/framework-guides/docs/framework/advances-configuration.md:32]: will just google.com work? or should it be http://google.com?

**Review by MPTG94 — APPROVED** [review]
Fix comments and merge

File: docs/framework-guides/docs/deployment/helm.md (hunk 1 above)
Comment (MatanHeledPort) [reply docs/framework-guides/docs/deployment/helm.md:67]: This is done for extra suspense

File: docs/framework-guides/docs/deployment/helm.md [hunk 6]
```diff
@@ -62,3 +62,24 @@ the [integration configuration](../develop-an-integration/integration-configurat
 More information about the available event listener types and optional configurations can be found in
 the [event listeners](../framework/features/event-listener.md) guide.
 :::
+
+## Advanced configruation
+The Ocean framework supports [advanced configuration](../framework/advances-configuration.md) using environment variables. The Ocean Helm chart allows setting these variables using Helm parameters. This can be done in one of two ways:
```
Comment (MPTG94) [inline docs/framework-guides/docs/deployment/helm.md]: URL here is wrong

File: docs/framework-guides/docs/deployment/helm.md [hunk 7]
```diff
@@ -62,3 +62,24 @@ the [integration configuration](../develop-an-integration/integration-configurat
...
+	--set extraEnv[0].name=HTTP_PROXY \
+	--set extraEnv[0].value=http://my-proxy.com:1111
+```
+
+2. The Helm `values.yaml` file:
+```yaml showLineNumbers
+# The rest of the configuration
+# ...
+extraEnvs:
+	- name: HTTP_PROXY
```
Comment (MPTG94) [inline docs/framework-guides/docs/deployment/helm.md:83]: This seems like quite a lot of indentation

Comment (MPTG94) [reply docs/framework-guides/docs/deployment/helm.md:83]: Also for some reason the red blocks are visible when viewing the file

**Review by MPTG94 — APPROVED** [review]
Some more small comments

**Pull request merged**